### PR TITLE
Avoid requesting an Avatar image bigger than what the template asks for

### LINKED ISSUE #16287: User avatar in profile page changed from s=290 to s=580 while rendered at s=128 anyway
Gitea 1.14.3 is requesting Libravatar avatar images with size 580 in profile page while rendering them at 128x128.
Older versino 1.12.3 was requesting a size of 290 instead.

I think it would make sense to request s=128 instead, to reduce the download time

### PR DESCRIPTION
**Pull request by strk**
Hopefully fixes #16287

It makes NO sense to request an avatar of 1160x1160 when it is rendered MUCH MUCH smaller.
See https://try.gitea.io/strk and right-click -> open image in new tab, then check the URL, is formed with `?size=1160` now. I think this PR makes it ask for a size=256 instead (default 128, sizeFactor 2)

### CODE REVIEW

File: models/avatars/avatar.go
Comment (wxiaoguang) [inline models/avatars/avatar.go]: This const shouldn't be changed. It is used for small avatars.

File: models/avatars/avatar.go
Comment (wxiaoguang) [inline models/avatars/avatar.go]: This one seems fine. L-G-T-M.

Comment (lunny) [reply models/avatars/avatar.go]: But I think it just has been changed in #15941 from 2 to 4. @silverwind

**Comment by zeripath** [comment]
@strk https://github.com/strk/gitea/pull/1  makes this configurable.

**Comment by strk** [comment]
Thanks @zeripath - I've merged your PR

**Review by zeripath — APPROVED** [review]
I'm going to approve this as it's now configurable. 

I don't mind which setting is used but 4 is almost certainly too high and 1 may be too low.

However, in the interests of getting this merged I will approve.

**Comment by lafriks** [comment]
Can we make default more close to current behavior, for example `3`?

**Comment by silverwind** [comment]
Yes, default should be `3` for reasons outlined in https://github.com/go-gitea/gitea/pull/17951.

**Comment by zeripath** [comment]
I put a pr against @silverwind's to add configuration - I'd be happy to approve that or if you want to make the suggestions against this pr to make the default 3 I think that would be acceptable to.

**Comment by silverwind** [comment]
https://github.com/go-gitea/gitea/pull/17951 is ready again.

**Review by wxiaoguang — APPROVED** [review]

**Review by silverwind — CHANGES_REQUESTED** [review]
Blocking merge as the consensus seems to be factor 3 is a good default. https://github.com/go-gitea/gitea/pull/17951 replaces this PR.

**Comment by wxiaoguang** [comment]
Oh, I see the different, then we use #17951  😊

**Comment by strk** [comment]
Thanks guys !

**Pull request closed without merging**
